src/lib/ngsi10/NotifyContextRequest.cpp

```diff
@@ -80,20 +80,34 @@ std::string NotifyContextRequest::toJson
   std::map<std::string, std::string>* replacementsP
 )
 {
-  if ((renderFormat != NGSI_V2_NORMALIZED) && (renderFormat != NGSI_V2_KEYVALUES) && (renderFormat != NGSI_V2_VALUES))
+  if ((renderFormat != NGSI_V2_NORMALIZED) && (renderFormat != NGSI_V2_KEYVALUES) && (renderFormat != NGSI_V2_VALUES) && (renderFormat != NGSI_V2_SIMPLIFIEDKEYVALUES) && (renderFormat != NGSI_V2_SIMPLIFIEDNORMALIZED))
   {
     OrionError oe(SccBadRequest, "Invalid notification format");
     alarmMgr.badInput(clientIp, "Invalid notification format");
 
     return oe.toJson();
-  }  
-
-  JsonObjectHelper jh;
-
-  jh.addString("subscriptionId", subscriptionId.get());
-  jh.addRaw("data", contextElementResponseVector.toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP));
+  }
+  else if (renderFormat == NGSI_V2_SIMPLIFIEDNORMALIZED)
+  {
+    std::string out;
+    out += contextElementResponseVector[0]->toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP);
+    return out;
+  }
+  else if (renderFormat == NGSI_V2_SIMPLIFIEDKEYVALUES)
+  {
+    renderFormat = NGSI_V2_KEYVALUES;
+    std::string out;
+    out += contextElementResponseVector[0]->toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP);
+    return out;
```

Comment:
Although contextElementResponseVector should contain exactly one entity, I'd suggest to do some "defensive programming" logic here so a bug in CB logic that for some reason provides an empty contextElementResponseVector vector doesn't crash the CB here.

Something like this:

```suggestion
    renderFormat = NGSI_V2_KEYVALUES;
    if (contextElementResponseVector.size() == 0)
    {
      LM_E(("Runtime Error (contextElementResponser MUST NOT be zero length)"));
      return "{}";
    }
    else
    {
      std::string out;
      out += contextElementResponseVector[0]->toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP);
      return out;
    }
```

The same also in L93 part.

Reply:
Updated code in https://github.com/telefonicaid/fiware-orion/pull/4298/commits/4bef2c1f3f62fb09d538271ff8e0554377b48d13